python/cuml/manifold/t_sne.pyx

```diff
@@ -199,6 +199,15 @@ class TSNE(Base,
         'sqeuclidean' metric, the distances will still be squared when True.
         Note: This argument should likely be set to False for distance metrics
         other than 'euclidean' and 'l2'.
+    precomputed_knn : array / sparse array / tuple, optional (device or host)
+        Either one of :
+            - Tuple (distances, indices) of arrays of
```

Comment:
We should consider also supporting the `knn_search_index` option (third tuple element) which is supported in the reference implementation: https://github.com/lmcinnes/umap/blob/3f19ce19584de4cf99e3d0ae779ba13a57472cd9/umap/umap_.py#L1626. We can push that feature off but we should at least create an issue for it to keep it on our radar.

Reply:
Sure, I think it would be better to work on this in a follow-up PR. Just opened an issue over here : https://github.com/rapidsai/cuml/issues/5118.